src/main/scala/com/sksamuel/scapegoat/inspections/VariableShadowing.scala

```diff
@@ -16,17 +15,17 @@ class VariableShadowing
 
   def inspector(context: InspectionContext): Inspector =
     new Inspector(context) {
-      override def postTyperTraverser =
+      override def postTyperTraverser: context.Traverser =
         new context.Traverser {
 
           import context.global._
 
-          private val contexts = new mutable.Stack[ListBuffer[String]]()
+          private var contexts: List[ListBuffer[String]] = Nil
```

Comment:
A `var List` is better than a `mutable.Stack`, so replacing it. Behavior is kept.